src/huggingface_hub/inference_api.py

```diff
@@ -0,0 +1,145 @@
+    """Client to configure requests and make calls to the HuggingFace Inference API.
+
+    Example:
+
+            >>> from huggingface_hub.inference_api import InferenceApi
+
+            >>> # Mask-fill example
+            >>> api = InferenceApi("bert-base-uncased")
+            >>> api(inputs="The goal of life is [MASK].")
+            >>> >> [{'sequence': 'the goal of life is life.', 'score': 0.10933292657136917, 'token': 2166, 'token_str': 'life'}]
+
+            >>> # Question Answering example
+            >>> api = InferenceApi("deepset/roberta-base-squad2")
+            >>> inputs = {"question":"What's my name?", "context":"My name is Clara and I live in Berkeley."}
+            >>> api(inputs)
+            >>> >> {'score': 0.9326569437980652, 'start': 11, 'end': 16, 'answer': 'Clara'}
+
+            >>> # Zero-shot example
+            >>> api = InferenceApi("typeform/distilbert-base-uncased-mnli")
+            >>> inputs = "Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!"
+            >>> params = {"candidate_labels":["refund", "legal", "faq"]}
+            >>> api(inputs, params)
+            >>> >> {'sequence': 'Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!', 'labels': ['refund', 'faq', 'legal'], 'scores': [0.9378499388694763, 0.04914155602455139, 0.013008488342165947]}
+
+            >>> # Overriding configured task
+            >>> api = InferenceApi("bert-base-uncased", task="feature-extraction")
+    """
+
+    def __init__(
+        self,
+        repo_id: str,
+        task: Optional[str] = None,
+        token: Optional[str] = None,
+        gpu: Optional[bool] = False,
+    ):
+        """Inits headers and API call information.
+
+        Args:
+            repo_id (``str``): Id of repository (e.g. `user/bert-base-uncased`).
+            task (``str``, `optional`, defaults ``None``): Whether to force a task instead of using task specified in the repository.
+            token (:obj:`str`, `optional`):
+                The API token to use as HTTP bearer authorization. This is not the authentication token.
+                You can find the token in https://huggingface.co/settings/token. Alternatively, you can
+                find both your organizations and personal API tokens using `HfApi().whoami(token)`.
+            gpu (``bool``, `optional`, defaults ``False``): Whether to use GPU instead of CPU for inference(requires Startup plan at least).
+        .. note::
+            Setting :obj:`token` is required when you want to use a private model.
+        """
+        self.options = {"wait_for_model": True, "use_gpu": gpu}
+
+        self.headers = {}
+        if isinstance(token, str):
+            self.headers["Authorization"] = "Bearer {}".format(token)
+
+        # Configure task
+        model_info = HfApi(ENDPOINT_HUB).model_info(repo_id=repo_id, token=token)
```

Comment:
Does this need to be hardcoded? That's already the default value of `HfApi`

Reply:
It was unclear to me, but without specifying it, the tests were using staging when running with GitHub Actions.

Reply:
You can use the `@with_production_testing` decorator on the tests you want to run in production settings

Reply:
Ah this wasn't here before, thanks!